hs_tracking/views.py

```diff
@@ -22,6 +24,28 @@ def get(self, request, **kwargs):
         querydict = dict(request.GET)
         url = querydict.pop('url')[0]
 
+        if settings.GEOHUB_HOMEPAGE_URL in url:
```

Comment:
As before I am concerned that here, and other places we should not be writing custom code for a specific app as part of the general hydroshare website.  Rather we should be achieving this functionality through launching the app in a standard way and having it call an API function that triggers the pushing of files into a separate iRODS zone.  I do not think that it is a sustainable design pattern to look for a specific URL and then have code that does specific custom work based on that.  If the URL hosting the app changes, then we will need code or parameter changes.  It is better to work with URLs that come from the web app resource instance and more general purpose API to achieve what you are trying to do.

Reply:
We could also implement hooks for custom code to be run for individual apps (a before open app event, that is called at the beginning of this method).  That way, the app won't have to make calls to hydroshare and the code could be structured in a more maintainable way.

Reply:
@sblack-usu That sounds like a good idea. I'll try to implement a hook for this and then ask you to review.